cmd/libsnap-confine-private/cgroup-support.c

```diff
@@ -80,6 +80,7 @@ static const char *cgroup_dir = "/sys/fs/cgroup";
 // hybrid or legacy) The algorithm is described in
 // https://systemd.io/CGROUP_DELEGATION/
 bool sc_cgroup_is_v2(void) {
+    bool hide_warning = getenv_bool("SNAPD_HIDE_CGROUPV2_WARNING", false);
```

Comment:
Would it be simpler to not add an additional `hide_warning` variable and instead just initialise `did_warn` with the result of `getenv_bool()`?

ie.

```c
static bool did_warn = getenv_bool("SNAPD_HIDE_CGROUPV2_WARNING", false);
```

Reply:
Thanks for the security review!
That would be simpler, but we can only initialize the `static bool did_warn` variable with a constant value, which the `getenv_bool()` function isn't. (It would throw an `initializer element is not constant` error otherwise.)